gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -684,6 +743,187 @@ public Boolean call() {
     }
   }
 
+  @Override
+  public Address get(final AddressId addressId, AddressOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Address answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Address>() {
+            @Override
+            public com.google.api.services.compute.model.Address call() {
+              switch (addressId.type()) {
+                case REGION:
+                  RegionAddressId regionAddressId = (RegionAddressId) addressId;
```

Comment:
I think we better make `AddressId` an abstract class with package scope constructor, that should solve all of these problems right? I'll just remove `RegionResourceId` and `ZoneResourceId`.